sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -915,7 +924,11 @@ class Analyzer(
               AnalysisContext.get.relationCache.getOrElseUpdate(
                 key, v1SessionCatalog.getRelation(v1Table.v1Table))
             case table =>
-              DataSourceV2Relation.create(table)
+              DataSourceV2Relation.create(
+                table,
+                catalogManager.catalogIdentifier(catalog),
```

Comment:
Why are we using "identifier" for the catalog name? Everywhere else we call it the catalog name, so I don't see a reason to make this more complicated.

Reply:
Sure. I was thinking about making the naming convention consistent since it's called identifier for tables. I'm totally okay with `CatalogName`.